piff/input.py

```diff
@@ -595,6 +619,14 @@ def _makeStarsFromImage(image_kwargs, cat_kwargs, wcs, chipnum,
                 logger.warning("Skipping this star.")
                 continue
 
+            # here we remove stars that have been at least partially covered by a mask
+            # and thus have weight exactly 0 in at least a certain number of pixels of their postage stamp
+            if max_mask_pixels is not None:
+                n_masked = wt_stamp.array.shape[0] * wt_stamp.array.shape[1] - np.count_nonzero(wt_stamp.array)
+                if n_masked >= max_mask_pixels:
+                    logger.warning("Star at position %f,%f has %i masked pixels.",x,y,n_masked)
```

Comment:
Probably ought to say that it is cut because of this.  Otherwise, this just sounds informative.